src/main/java/org/folio/helper/BindHelper.java

```diff
@@ -55,26 +64,54 @@ public Future<ReceivingResults> bindPieces(BindPiecesCollection bindPiecesCollec
   }
 
   private Future<ReceivingResults> processBindPieces(BindPiecesCollection bindPiecesCollection, RequestContext requestContext) {
-    // 1. Get piece records from storage
+    //   1. Get piece records from storage
     return retrievePieceRecords(requestContext)
-      // 2. Update piece isBound flag
+      // 2. Check if there are any outstanding requests for items
+      .compose(piecesGroupedByPoLine -> checkRequestsForPieceItems(piecesGroupedByPoLine, bindPiecesCollection, requestContext))
+      // 3. Update piece isBound flag
       .map(this::updatePieceRecords)
-      // 3. Update currently associated items
+      // 4. Update currently associated items
       .map(piecesGroupedByPoLine -> updateItemStatus(piecesGroupedByPoLine, requestContext))
-      // 4. Crate item for pieces with specific fields
+      // 5. Crate item for pieces with specific fields
       .compose(piecesGroupedByPoLine -> createItemForPiece(piecesGroupedByPoLine, bindPiecesCollection, requestContext))
-      // 5. Update received piece records in the storage
+      // 6. Update received piece records in the storage
       .compose(piecesGroupedByPoLine -> storeUpdatedPieceRecords(piecesGroupedByPoLine, requestContext))
-      // 6. Update Title with new bind items
+      // 7. Update Title with new bind items
       .map(piecesGroupedByPoLine -> updateTitleWithBindItems(piecesGroupedByPoLine, requestContext))
-      // 7. Return results to the client
+      // 8. Return results to the client
       .map(piecesGroupedByPoLine -> prepareResponseBody(piecesGroupedByPoLine, bindPiecesCollection));
   }
 
+  private Future<Map<String, List<Piece>>> checkRequestsForPieceItems(Map<String, List<Piece>> piecesGroupedByPoLine, BindPiecesCollection bindPiecesCollection, RequestContext requestContext) {
+    var itemIds = extractAllPieces(piecesGroupedByPoLine)
+      .map(Piece::getItemId)
+      .filter(StringUtils::isNotEmpty)
+      .toList();
```

Comment:
you should have some map here where tenantId will be a key and list of itemIds would be a value, and for each tenantId make request inventoryItemRequestService.getItemsWithActiveRequests(itemIds, requestContext).
To make request in another tenant you should modify requestContext, here is method that can help to do it: RequestContextUtil::createContextWithNewTenantId

Reply:
Modified the methods to fetch items from different tenants